lib/extract/netcdf/extract_netcdf_base.jinja

```diff
@@ -0,0 +1,464 @@
+            stop
+        endif
+        CheckError = retval
+    end function CheckError
+
+    ! -------------------------------------------------------------------------
+    !> This is a one-time init function. It is not required for the kernel
+    !! extraction and is therefore empty.
+    subroutine extract_PSyDataInit()
+    end subroutine extract_PSyDataInit
+
+    ! -------------------------------------------------------------------------
+    !> This is a one-time shutdown function. It is not required for the kernel
+    !! extraction and is therefore empty.
+    subroutine extract_PSyDataShutdown()
+    end subroutine extract_PSyDataShutdown
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is the first function called when data is written out
+    !! before an instrumented region of code.
+    !! @param[inout] this The instance of the ExtractNetcdfBaseType.
+    !! @param[in] module_name The name of the module of the instrumented
+    !!            region.
+    !! @param[in] region_name The name of the instrumented region.
+    !! @param[in] num_pre_vars The number of variables that are declared and
+    !!            written before the instrumented region.
+    !! @param[in] num_post_vars The number of variables that are also declared
+    !!            before an instrumented region of code, but are written after
+    !!            this region.
+    subroutine PreStart(this, module_name, region_name, num_pre_vars, &
+                        num_post_vars)
+        use netcdf, only: nf90_create, NF90_CLOBBER
+        implicit none
+        class(ExtractNetcdfBaseType), intent(inout), target :: this
+        character(*), intent(in) :: module_name, region_name
+        integer, intent(in)      :: num_pre_vars, num_post_vars
+        integer :: retval
+
+        call this%PSyDataBaseType%PreStart(module_name, region_name, &
+                                           num_pre_vars, num_post_vars)
+
+        ! Open the NetCDF file
+        retval = CheckError(nf90_create(module_name//"-"//region_name//".nc", &
+                                        NF90_CLOBBER, this%ncid))
+        ! Allocate the array that will store the variable IDs of all
+        ! variables that are going to be declared:
+        allocate(this%var_id(num_pre_vars+num_post_vars))
+        
+    end subroutine PreStart
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is called to open a NetCDF file for reading. The
+    !! filename is based on the module and kernel name. This is used by a
+    !! driver program that will read a NetCDF file previously created by the
+    !! PSyData API.
+    !! @param[inout] this The instance of the ExtractNetcdfBaseType.
+    !! @param[in] module_name The name of the module of the instrumented
+    !!            region.
+    !! @param[in] kernel_name The name of the instrumented region.
```

Comment:
Would this be better as `region_name` to make it consistent with PreStart()?

Reply:
Indeed.